lib/api/core/internalStorage.js

```diff
@@ -0,0 +1,181 @@
+    return this.storageEngine.search(
+        this.index,
+        collection,
+        searchBody,
+        options);
+  }
+
+  /**
+   * Scroll results from previous search
+   *
+   * @param {String} collection - Collection name
+   * @param {String} scrollId - Scroll identifier
+   * @param {Object} options - scroll (default scrollTTL)
+   *
+   * @returns {Promise.<Object>} { scrollId, hits, total }
+   */
+  scroll (collection, scrollId, options) {
+    return this.storageEngine.scroll(
+      this.index,
+      collection,
+      scrollId,
+      options);
+  }
+
+  /**
+   * Creates a new document
+   *
+   * @param {String} collection - Collection name
+   * @param {Object} content - Document content
+   * @param {Object} options - id (null), refresh (false), userId (null)
+   *
+   * @returns {Promise.<Object>} { _id, _version, _source }
+   */
+  create (collection, content, options) {
+    return this.storageEngine.create(this.index, collection, content, options);
+  }
+
+  /**
+   * Creates a new document, or replace it if it already exist
+   *
+   * @param {String} collection - Collection name
+   * @param {String} id - Document id
+   * @param {Object} content - Document content
+   * @param {Object} options - refresh (false), userId (null)
+   *
+   * @returns {Promise.<Object>} { _id, _version, _source, created }
+   */
+  createOrReplace (collection, id, content, options) {
+    return this.storageEngine.createOrReplace(
+      this.index,
+      collection,
+      id,
+      content,
+      options);
+  }
+
+  /**
+   * Replaces a document
+   *
+   * @param {String} collection - Collection name
+   * @param {String} id - Document id
+   * @param {Object} content - Document content
+   * @param {Object} options - refresh (false), userId (null)
+   *
+   * @returns {Promise.<Object>} { _id, _version, _source }
+   */
+  replace (collection, id, content, options) {
+    return this.storageEngine.replace(
+      this.index,
+      collection,
+      id,
+      content,
+      options);
+  }
+
+  /**
+   * Partialy updates a document
+   *
+   * @param {String} index - Index name
+   * @param {String} collection - Collection name
+   * @param {String} id - Document id
+   * @param {Object} content - Updated content
+   * @param {Object} options - refresh (false), userId (null), retryOnConflict (0)
+   *
+   * @returns {Promise.<Object>} { _id, _version }
+   */
+  update (collection, id, content, options) {
+    return this.storageEngine.update(this.index, collection, id, content, options);
+  }
+
+  /**
+   * Deletes a document
+   *
+   * @param {String} collection - Collection name
+   * @param {String} id - Document id
+   * @param {Object} options - refresh (false), retryOnConflict (-1)
+   *
+   * @returns {Promise.<Object>} { _id }
+   */
+  delete (collection, id, options) {
+    return this.storageEngine.delete(this.index, collection, id, options);
+  }
+
+  // @todo createCollection + add to indexCache
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)